### ISSUE TITLE
Update to 5.0

### ISSUE THREAD

**Issue by bobmulder** [uafrica/oauth-server#35]
ThePHPLeague just released 5.x (https://github.com/thephpleague/oauth2-server/releases/tag/5.0.1).
It would be great to update to that version!

**Comment by dakota** [comment]
Looks like it requires PSR-7 compatible Request/Response objects which CakePHP 3 does not support yet. Should hopefully be in 3.3 though.

**Comment by bobmulder** [comment]
Good point! Will it be implemented when 3.3 supports PSR-7 compatible Request/Response objects?

**Comment by dakota** [comment]
Once we've upgraded to 3.3 internally we'll probably take a look :) Unless I get bored and take a look before then.

**Comment by skie** [comment]
Take a look into https://github.com/markstory/cakephp-spekkoek plugin.

**Comment by bobmulder** [comment]
Thanks @skie! I like the concept. Would be awesome if implemented in CakePHP!

**Comment by dakota** [comment]
Spekkoek is just the proof of concept, it is being merged into core piece by piece (https://github.com/cakephp/cakephp/pull/8696)

**Comment by sukihub** [comment]
Hi :) is there any work being done on this? Would you possibly accept pull requests?

**Comment by kevinchow** [comment]
My branch is using oAuth2 6.0, if you guys are interested I can submit a pull request.
P.S I haven't create any unit test cases yet.

**Comment by dakota** [comment]
@kevinchow Yes please :)

**Comment by kevinchow** [comment]
Okay, I will do some clean up before submit the pull request.
